### PR TITLE
fix: pass the missing `extra_hosts` argument

### LINKED ISSUE #6888: Bug: `sam local start-api --add-host` does not work
### Description:

I executed `sam local start-api --add-host="host.docker.internal:host-gateway"`, but it didn't reflect in the hosts file.

### Steps to reproduce:

Execute `sam local start-api --add-host="xxx"` command.

### Additional environment details (Ex: Windows, Mac, Amazon Linux etc)

1. OS: Ubuntu 22.04.3 LTS (5.15.146.1-microsoft-standard-WSL2)
2. `sam --version`:  SAM CLI, version 1.113.0
3. AWS region:

```
# Paste the output of `sam --info` here
$ sam --info
{
  "version": "1.113.0",
  "system": {
    "python": "3.11.3",
    "os": "Linux-5.15.146.1-microsoft-standard-WSL2-x86_64-with-glibc2.35"
  },
  "additional_dependencies": {
    "docker_engine": "24.0.7",
    "aws_cdk": "Not available",
    "terraform": "1.6.5"
  },
  "available_beta_feature_env_vars": [
    "SAM_CLI_BETA_FEATURES",
    "SAM_CLI_BETA_BUILD_PERFORMANCE",
    "SAM_CLI_BETA_TERRAFORM_SUPPORT",
    "SAM_CLI_BETA_RUST_CARGO_LAMBDA"
  ]
}
```

`Add --debug flag to command you are running`

### PR DESCRIPTION
**Pull request by gkuga**
#### Which issue(s) does this change fix?

This fixes #6888 and related to #6078.
I have confirmed that this change makes `--add-host` work.

#### Why is this change necessary?
I change to pass the `extra_hosts` that were not passed as arguments.
Because `sam local start-api --add-host` does not work.

#### How does it address the issue?
Pass the missing argument.

#### Mandatory Checklist
**PRs will only be reviewed after checklist is complete**

- [x] Add input/output [type hints](https://docs.python.org/3/library/typing.html) to new functions/methods
- [x] Write design document if needed ([Do I need to write a design document?](https://github.com/aws/aws-sam-cli/blob/develop/DEVELOPMENT_GUIDE.md#design-document))
- [x] Write/update unit tests
- [x] Write/update integration tests
- [x] Write/update functional tests if needed
- [x] `make pr` passes
- [x] `make update-reproducible-reqs` if dependencies were changed
- [x] Write documentation

By submitting this pull request, I confirm that my contribution is made under the terms of the [Apache 2.0 license](https://www.apache.org/licenses/LICENSE-2.0).

### CODE REVIEW

**Comment by lucashuy** [comment]
Thanks for making creating this fix! The changes look good to me, can you run a quick `make format` to fix any formatting issues in the source code? Looks like that is what is causing the PR jobs to fail at the moment.

**Comment by gkuga** [comment]
@lucashuy Thanks for your feedback. I've run `make format`. Will the workflow pass?

**Review by lucashuy — APPROVED** [review]

**Comment by lucashuy** [comment]
Thanks for updating the PR, looks like the workflow passes! Just need one more reviewer from our team before we can merge this in.

File: samcli/local/lambdafn/runtime.py
Comment (mndeveci) [inline samcli/local/lambdafn/runtime.py:71]: Thanks for updating missing documentation!

**Review by mndeveci — APPROVED** [review]

**Pull request merged**
